haystack/fields.py

```diff
@@ -153,7 +153,9 @@ def get_iterable_objects(cls, current_objects):
                 return current_objects.all()
             return []
 
-        elif not hasattr(current_objects, '__iter__'):
+        try:
+            _ = [obj for obj in current_objects]
```

Comment:
This will consume all the possible generators:

```
class Test(object):
    @property
    def test(self):
        for x in '12345': yield x

current_objects = Test().test
_ = [c for c in current_objects]
print(', '.join(current_objects))
```

Reply:
Yes — I think the way to handle this would need to be something like this:

```python
try:
    iter(current_objects)
except TypeError:
    current_objects = [current_objects]
```

Reply:
I hadn't realized this PR already existed, but I used the `iter(current_objects)` approach suggested by @acdha for PR #1663, back in 2019.